src/auto-evo/simulation/food_source/EnvironmentalFoodSource.cs

```diff
@@ -15,14 +15,16 @@ public EnvironmentalFoodSource(Patch patch, Compound compound, float foodCapacit
 
             this.patch = patch;
             this.compound = compound;
-            totalEnvironmentalEnergySource = patch.Biome.Compounds[this.compound].Ambient * foodCapacityMultiplier;
+            totalEnvironmentalEnergySource = compound.InternalName == "sunlight" ?
+                patch.Biome.Sunlight!.Average * foodCapacityMultiplier :
```

Comment:
This nullability override feels pretty wrong to me here, even though the check is above, I think this should be rewritten like the line below (access using the compound key).